### ISSUE TITLE
Mouseover checker doesn't update when moving or attacking (Part 7)

### ISSUE THREAD

**Issue by lnmiltz** [SelinaDev/Godot-Roguelike-Tutorial#28]
This is a really tiny problem but I figured it's worth submitting an issue

The mouseover label updates when the player moves the mouse, but if you move or perform an action like picking up an item or killing an enemy, it doesn't update.

The mouseover checker code looks like this in part 7:

```gdscript
func _unhandled_input(event: InputEvent) -> void:
    if event is InputEventMouseMotion:
        var mouse_position: Vector2 = get_local_mouse_position()
        var tile_position: Vector2i = Grid.world_to_grid(mouse_position)
        if _mouse_tile != tile_position:
            _mouse_tile = tile_position
            var entity_names = get_names_at_location(tile_position)
            entities_focused.emit(entity_names)
```

I don't know if there's a cleaner solution, but to solve it I moved all the code under the `if` to a new function `update_names_at_location()`. Then I called it here in game.gd's _physics_process:

```diff
+   @onready var mouseover_checker: MouseoverChecker = $Map/MouseoverChecker

    ...

    func _physics_process(_delta) -> void:
        ...
         _handle_enemy_turns()
        map.update_fov(player.grid_position)
+       mouseover_checker.update_names_at_location()
```

For some reason this causes it to start flickering, but I added `await get_tree().process_frame` to the start of the new function and it was fixed.

**Comment by SelinaDev** [comment]
Thank you for reporting this! And thank you for providing a solution right away. I'll have to see when I'll get a chance to integrate this, but I'll be sure to fix this.

**Comment by SelinaDev** [comment]
I just updated the code. I opted to move the code into into a `_process()` function right in the MouseoverChecker node. I also removed the caching of the tile position. It's not optimal, but it does what it needs to do (and doesn't seem to impact the performance).
I think a better solution would be to only call this code in reaction to either the player taking a turn (e.g., triggered by a signal from the event handler) or the player moving the mouse. I'm leaving this here as a hint if anyone wants to optimize like that, but I don't want to change the tutorial too much at this point, so I'm sticking to this suboptimal but working solution.
I'll update the text of the tutorial and upload that shortly.
